WalletWasabi/WabiSabi/Client/RoundStateUpdater.cs

```diff
@@ -26,8 +27,20 @@ public RoundStateUpdater(TimeSpan requestInterval, IWabiSabiApiRequestHandler ar
 
 	protected override async Task ActionAsync(CancellationToken cancellationToken)
 	{
-		var statusResponse = await ArenaRequestHandler.GetStatusAsync(cancellationToken).ConfigureAwait(false);
-		RoundStates = statusResponse.ToDictionary(round => round.Id);
+		var request = new RoundStateRequest(RoundStates.Select(x => (x.Key, x.Value.CoinjoinState.Events.Count)).ToImmutableList());
+		RoundState[] statusResponse = await ArenaRequestHandler.GetStatusAsync(request, cancellationToken).ConfigureAwait(false);
+
+		var updatedRoundStates = statusResponse
+			.Where(rs => RoundStates.ContainsKey(rs.Id))
+			.Select(rs => (Update: rs, CurrentRoundState: RoundStates[rs.Id]))
+			.Select(x => x.Update with {
+				CoinjoinState = x.Update.CoinjoinState.MergeBack(x.CurrentRoundState.CoinjoinState)
+			}).ToList();
+
+		var newRoundStates = statusResponse
+			.Where(rs => !RoundStates.ContainsKey(rs.Id));
+
+		RoundStates = newRoundStates.Concat(updatedRoundStates).ToDictionary(x => x.Id, x => x);
```

Comment:
The missing rounds should be removed.

Reply:
They are because  `round-states = new-states + updated-states`.

Reply:
ACK!